openstack/utils/templates/_helpers.tpl

```diff
@@ -60,3 +60,25 @@ enabled = false
     {{- $options := merge .Values.osprofiler .Values.global.osprofiler }}
     {{- if $options.enabled }} osprofiler{{ end -}}
 {{- end }}
+
+{{- define "jaeger_agent_sidecar" }}
+- image: jaegertracing/jaeger-agent:{{ .Values.osprofiler.jaeger.version }}
+  name: jaeger-agent
+  ports:
+    - containerPort: 5775
+      name: zk-compact-trft
+      protocol: UDP
+    - containerPort: 6831
+      name: jg-compact-trft
+      protocol: UDP
+    - containerPort: 6832
+      name: jg-binary-trft
+      protocol: UDP
+    - containerPort: 5778
+      name: config-rest
+      protocol: TCP
+    - containerPort: 14271
+      name: admin-http
+      protocol: TCP
+  args: ["--collector.host-port=openstack-jaeger-collector.jaeger-infra.svc:14267"]
+{{- end }}
```

Comment:
a helper function to define the jaeger agent side car container

Reply:
If we have versioning constraints, does that mean we have to update every service using jaeger if we update the jaeger operator immediately?